server/tests/common/client.py

```diff
@@ -724,6 +724,172 @@ async def shamirorg(
         await testbed.drop_organization(organization_id)
 
 
+@dataclass(slots=True)
+class WorkspaceHistoryOrgRpcClients:
+    """
+    WorkspaceHistory contains:
+    - 3 users: `alice` (admin), `bob` (regular) and `mallory` (regular)
+    - 3 devices: `alice@dev1`, `bob@dev1` and `mallory@dev1` (every device has its local
+      storage up-to-date regarding the certificates)
+    - On 2001-01-01, Alice creates and bootstraps workspace `wksp1`
+    - On 2001-01-02, Alice uploads workspace manifest in v1 (empty)
+    - On 2001-01-03T01:00:00, Alice uploads file manifest `bar.txt` in v1 (containing "Hello v1")
+    - On 2001-01-03T02:00:00, Alice uploads folder manifest `foo` in v1 (empty)
+    - On 2001-01-03T03:00:00, Alice uploads workspace manifest in v2 (containing `foo` and `bar.txt`)
+    - On 2001-01-04, Alice gives Bob access to the workspace (as OWNER)
+    - On 2001-01-05, Alice gives Mallory access to the workspace (as CONTRIBUTOR)
+    - On 2001-01-06, Bob removes access to the workspace from Alice
+    - On 2001-01-07, Mallory uploads file manifest `bar.txt` in v2 (containing "Hello v2 world", stored on 2 blocks)
+    - On 2001-01-08T01:00:00, Mallory uploads folder manifest `foo` in v2 (containing `spam` and `egg.txt`)
+    - On 2001-01-08T02:00:00, Mallory uploads file manifest `foo/egg.txt` in v1 (empty file)
+    - On 2001-01-08T03:00:00, Mallory uploads folder manifest `foo/spam` in v1 (empty)
+    - On 2001-01-09, Bob uploads workspace manifest in v3 with renames `foo`->`foo2` and `bar.txt`->`bar2.txt`
+    - On 2001-01-10, Bob uploads file manifest `bar.txt` (now named `bar2.txt`) in v3 with content "Hello v3"
+    - On 2001-01-11, Bob uploads file manifest `foo/egg.txt` in v2 with content "v2"
+    - On 2001-01-12, Bob uploads folder manifest `foo` (now named `foo2`) in v3 with rename `egg.txt`->`egg2.txt` and removed `spam`
+    - On 2001-01-30, Bob gives back access to the workspace to Alice (as READER)
+
+    In the end we have the following per-entry history:
+      - `/`
+        - 2001-01-02: v1 from Alice, content: []
+        - 2001-01-03T03:00:00: v2 from Alice, content: ["foo", "bar.txt"]
+        - 2001-01-09: v3 from Bob, content: ["foo2", "bar2.txt"]
+    - `bar.txt`:
+      - 2001-01-03T01:00:00: v1 from Alice, content: "Hello v1"
+      - 2001-01-07: v2 from Alice, content: "Hello v2 world"
+      - 2001-01-10: v3 from Bob, content "Hello v3"
+    - `foo`:
+      - 2001-01-03T02:00:00: v1 from Alice, content: []
+      - 2001-01-08T01:00:00: v2 from Mallory, content ["spam", "egg.txt"],
+        with children not existing themselves yet !
+      - 2001-01-12: v3 from Bob, content ["egg2.txt"]
+    - `foo/egg.txt`:
+      - 2001-01-08T02:00:00: v1 from Mallory, content ""
+      - 2001-01-11: v2 from Bob, content "v2"
+    - `foo/spam`:
+      - 2001-01-08T03:00:00: v1 from Mallory, content: []
+
+    See `libparsec/crates/testbed/src/templates/workspace_history.rs` for it actual definition.
+    """
```

Comment:
Why not leaving just the link to the actual definition? 

I agree is handy to have the definition here, but IMO the cost of duplication is much greater than the "burden" of just opening two files next to each other in your IDE.

```suggestion
    """
    The organization details are described in:
    `libparsec/crates/testbed/src/templates/workspace_history.rs`
    """
```